Aur 315 get signed verification by brightid

Bob-six opened the pull request.

Bob-six committed:
feat: added a new route for verifiying user by brightid

Bob-six committed:
fix: added testcases and fix type errors

Bob-six committed:
feat: updated the zip files

adamstallard commented on web_services/foxx/brightid/index.js:
```diff
@@ -722,7 +864,44 @@ router
+  .response(null)
+
+router
+  .get("/verifications/BrightID/:app/:brightId", handlers.createSignatureById)
```
Why did you create a new route for this when [the issue](https://linear.app/aura-brightid/issue/AUR-315/get-signed-verification-by-brightid) says to reuse the existing route, and if the appUserID isn't found, then look for the BrightID? Were you worried that the DB query would be too expensive?


> Using the same url structure as the /verification endpoint for getting a signed verification for an appUserId/ContextId, fallback to checking a BrightID also.

Bob-six replied:
It was so complex at first to implement it inside the same function. I forgot to make it inside the same route after it was completed. I will make it as we disscussed on the issue

Bob-six committed:
fix: reuse the existing route for generating the signature instead

Bob-six committed:
chore: add propper formatting to the project

Bob-six committed:
chore: add editor config to sync up with prettier config

adamstallard commented on web_services/foxx/brightid/index.js:
```diff
@@ -658,198 +807,238 @@ router
+  )
+  .response(schemas.verificationsGetResponse)
+  .error(404, 'brightid not found');
+
```
Do we still want this route?

Bob-six replied:
I removed this route

adamstallard commented on web_services/foxx/brightid/index.js:
```diff
@@ -210,11 +210,147 @@ const handlers = {
+        let message, h;
+        if (app.idsAsHex) {
+          message = pad32(appKey) + addressToBytes32(userId);
+        } else {
+          if (userId.length > 32) {
+            throw new errors.UnsingableAppUserIdError(userId);
+          }
+          message = pad32(appKey) + pad32(userId);
+        }
+        message = Buffer.from(message, 'binary').toString('hex');
+        if (includeHash) {
+          message += verificationHash;
+        }
+        if (timestamp) {
+          const t = timestamp.toString(16);
+          message += '0'.repeat(64 - t.length) + t;
+        }
+        h = new Uint8Array(
+          createKeccakHash('keccak256').update(message, 'hex').digest()
+        );
+        publicKey = ethKeyPair.publicKey;
+        const _sig = secp256k1.ecdsaSign(h, ethKeyPair.privateKey);
+        sig = {
+          r: Buffer.from(Object.values(_sig.signature.slice(0, 32))).toString(
+            'hex'
+          ),
+          s: Buffer.from(Object.values(_sig.signature.slice(32, 64))).toString(
+            'hex'
+          ),
+          v: _sig.recid + 27,
+        };
+      }
+
+      result['sig'] = sig;
+      result['publicKey'] = publicKey;
+      results.push(result);
+    }
+
+    res.send({ data: results });
+  },
```
Almost all of this function is already contained in `verificationsGet()`

Bob-six replied:
I've made 2 functions inside this file and replaced the duplicate codes with the function calls. and checked the tests

Bob-six committed:
fix: update and refactor based on pr recommendations

Bob-six committed:
chore: update zip files

adamstallard commented on web_services/foxx/brightid/index.js:
```diff
@@ -210,11 +316,90 @@ const handlers = {
+
+    let userId = req.param('appUserId');
+
+    const vel = app.verificationExpirationLength;
+    const roundedTimestamp = vel ? parseInt(Date.now() / vel) * vel : 0;
+
+    if (timestamp == 'seconds') {
+      timestamp = vel ? roundedTimestamp / 1000 : parseInt(Date.now() / 1000);
+    } else if (timestamp == 'milliseconds') {
+      timestamp = vel ? roundedTimestamp : Date.now();
+    } else {
+      timestamp = undefined;
+    }
```
This code to get the timestamp from the vel was already duplicated twice (not your fault) but now it's duplicated three times. Can you try to refactor it to remove the duplication?

adamstallard committed:
Update version to 6.18 in manifest.json

Let's bump the version for this new release

adamstallard committed:
Change version to 6.17.3

6.18 is being used by regular BrightID nodes, so we can't upgrade to that until we've merged all the upstream code
